packages/functions/dev/src/registry.ts

```diff
@@ -100,7 +101,14 @@ export class FunctionsRegistry {
     this.handleEvent = eventHandler ?? (() => {})
     this.internalFunctionsPath = internalFunctionsPath
     this.projectRoot = projectRoot
-    this.timeouts = timeouts
+
+    // Calculate timeouts from config if not provided as override
+    const siteTimeout = config?.siteInfo?.functions_timeout ?? config?.siteInfo?.functions_config?.timeout
+    this.timeouts = {
+      syncFunctions: timeouts?.syncFunctions ?? siteTimeout ?? SYNCHRONOUS_FUNCTION_TIMEOUT,
+      backgroundFunctions: timeouts?.backgroundFunctions ?? siteTimeout ?? BACKGROUND_FUNCTION_TIMEOUT,
```

Comment:
In CLI background functions only get default without using `siteInfo` https://github.com/netlify/cli/blob/86c8598399168378e35ce5cb54f00b556dc2c091/src/utils/dev.ts#L142-L145

I'm not sure what the correct behavior is, just flagging this being different than CLI. Docs don't say anything about configurable timeout for background functions in https://docs.netlify.com/build/functions/background-functions/ , but I also had trouble finding information about this for regular NFs, so I'm not sure if docs are good source of desired behavior to replicate